app/models/miq_task.rb

```diff
@@ -295,6 +296,17 @@ def process_cancel
     end
   end
 
+  def attributes_hash_when_state_change(state, attributes = {})
+    case attributes[:state] = state
```

Comment:
I'm not a fan of mixing an assignment and a case at the same time.  These can just be two separate lines and it would be clear.